Symfony/CS/Fixer/Contrib/UselessReturnFixer.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class UselessReturnFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $functionOpen = $tokens->getNextTokenOfKind($index, array('{'));
+            $this->fixFunction($tokens, $functionOpen, $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $functionOpen));
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'These should not be an empty return statement at the end of a function.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run before ExtraEmptyLinesFixer, WhitespacyLinesFixer and after EmptyReturnFixer
+        return -18;
+    }
+
+    private function fixFunction(Tokens $tokens, $start, $end)
+    {
+        for ($index = $end; $index > $start; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_RETURN)) {
+                continue;
+            }
+
+            $nextAt = $tokens->getNextMeaningfulToken($index);
+            if (!$tokens[$nextAt]->equals(';')) {
+                continue;
+            }
+
+            if ($tokens->getNextMeaningfulToken($nextAt) !== $end) {
+                continue;
+            }
+
+            $this->removeReturnStatement($tokens, $index, $nextAt);
+        }
+    }
+
+    private function removeReturnStatement(Tokens $tokens, $start, $semiColonIndex)
+    {
+        for ($index = $semiColonIndex; $index >= $start; --$index) {
+            if ($tokens[$index]->isComment()) {
+                continue; // preserve comments
+            }
+
+            $tokens[$index]->clear();
+        }
+
+        // merge whitespace tokens if needed
+        if ($tokens[$index]->isWhitespace()) {
+            $content = $tokens[$index]->getContent();
+            for ($i = $index + 1; ; ++$i) {
+                if (!$tokens[$i]->isWhitespace()) {
+                    break;
+                }
+                $content .= $tokens[$i]->getContent();
+                $tokens[$i]->clear();
```

Comment:
You need to merge 2 T_WHITESPACE being placed one after another. Where could be the situation with more than 2 ? I think this loop is not needed.